src/gui/mainwindow.cpp

```diff
@@ -126,6 +117,7 @@ MainWindow::MainWindow(QWidget *parent)
 #if defined(Q_OS_WIN) || defined(Q_OS_MAC)
     , m_wasUpdateCheckEnabled(false)
 #endif
+    , m_searchEngine(new SearchEngine)
```

Comment:
To immediately prevent unnecessary questions and objections, I'll tell you that I even understand why you did it: you need to connect to its (SearchEngine) signals. There are (generally) two options: have a permanent instance (as you did) or to track its creation in any way (this is not possible in the current design).
But the problem here is different. When the current design has weaknesses preventing implement some feature correctly, we can choose one of two things: either fix the design or further degrade it. Unfortunately, you chose the latter.